src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,262 @@
+   * @param Image Given Input Image
+   * @param loc Locations at which features need to be extracted
+   * @param RegFtr Output the Regular Features
+   * @param SSFtr Output the Self Similarity Features
+   * @param table a helper vector required to convert image to LAB space.
+   */
+  void GetFeatures(const MatType &Image, arma::umat &loc,
+                   CubeType& RegFtr, CubeType& SSFtr,
+                   const arma::vec& table);
+
+  /**
+   * Makes a reflective border around an Image.
+   * @param InImage, Image which we have to make border around.
+   * @param top, border length (to be incremented) at top.
+   * @param left, border length at left.
+   * @param bottom, border length at bottom.
+   * @param right, border length at right.
+   * @param OutImage, Output Image. 
+   */  
+  void CopyMakeBorder(const CubeType& InImage, size_t top, 
+                 size_t left, size_t bottom, size_t right,
+                 CubeType& OutImage);
+  
+  /**
+   * Augment image patch with multiple channels of information 
+   * resulting in a feature vector. Shrink these channels by shrink
+   * size to reduce dimensions of the extracted candidate features.
+   * Refer to the following paper for details:
+   * Fast edge detection using structured forests
+   * Authors: Piotr Dollar, Larry Zitnick
+   * Published In: ICCV
+   * @param InImage Input Image.
+   * @param regCh Channels used in calculating Regular Features.
+   * @param ssCh Channels used in calculating Self Similarity Features.
+   * @param table a vector which helps in converting image from RGB to LUV.
+   */
+  void GetShrunkChannels(const CubeType& InImage, CubeType &reg_ch,
+                  CubeType &ss_ch, const arma::vec& table);
+  
+  /**
+   * Converts an Image in RGB color space to LUV color space.
+   * RGB values must be normalized i.e., range in [0.0, 1.0].
+   * @param InImage Input Image in RGB color space.
+   * @param OutImage Ouptut Image in LUV color space.
+   */
+  void RGB2LUV(const CubeType& InImage, CubeType& OutImage,
+                   const arma::vec& table);
+  
+  /**
+   * Resizes the Image to the given size using Bilinear Interpolation
+   * @param src Input Image
+   * @param height Height of Output Image.
+   * @param width Width Out Output Image.
+   * @param dst Output Image resized to (height, width)
+   */  
+  void BilinearInterpolation(const MatType& src,
+                            const size_t height, 
+                            const size_t width,
+                            MatType& dst);
+  
+  /**
+   * Applies a separable linear filter to an Image
+   * @param InOutImage Input/Output Contains the input Image, The final filtered Image is
+   *          stored in this param.
+   * @param kernel Input Kernel vector to be applied on Image.
+   * @param radius amount, the Image should be padded before applying filter.
+   */  
+  void Convolution(CubeType &InOutImage, const MatType& Filter, const size_t radius);
```

Comment:
Another picky comment, please use prefix const, also I know it's a matter of style, but I prefer to use type& parameter instead of type &parameter. For more details: https://github.com/mlpack/mlpack/pull/523/files/c92dd2fc6561b1cf40efb744dcc9233bc3675c91#r60094845